Add documentation on security features - take 2 (#3986)
theforeman/foreman-documentation: guides/common/modules/con_security-considerations.adoc

> [id="security-considerations"]

Reviewer: ```suggestion
:_mod-docs-content-type: CONCEPT

[id="security-considerations"]
```

[id="security-considerations"]
= Security considerations in {Project}

> When deploying {Project}, consider implementing security features that enhance the overall security of your system.

Reviewer 2: Is enabling more appropriate than implementing?

[NOTE]
====
These security considerations apply to both {ProjectServer} and {SmartProxyServer} environments.
====

> 

Reviewer: ```suggestion
```

Federal Information Processing Standards (FIPS) Mode::
{Project} supports running on FIPS-enabled {RHEL} hosts to comply with security standards for cryptographic modules. 

> However, {Project} itself is not FIPS-certified. In addition, you cannot enable FIPS mode after the installation of {Project}.

Reviewer: ```suggestion
However, {Project} itself is not FIPS-certified.
In addition, you cannot enable FIPS mode after the installation of {Project}.
```

+
FIPS mode enforces the use of validated cryptographic algorithms and modules.
Enabling FIPS mode ensures that all cryptographic operations within the system adhere to strict security standards.
+
For more information, see {RHELDocsBaseURL}9/html/security_hardening/switching-rhel-to-fips-mode_security-hardening[Switching RHEL to FIPS mode] in _{RHEL}{nbsp}9 Security hardening_.
+
[NOTE]
====
{Project} supports DEFAULT and FIPS crypto-policies.

> The FUTURE crypto-policy is not supported for {Project} and {SmartProxy} installations.

Reviewer: ```suggestion
The FUTURE crypto-policy is not supported for {ProjectServer} and {SmartProxyServer} installations.
```

====

File Access Policy Daemon (`fapolicyd`)::
By enabling `fapolicyd` on your {ProjectServer}, you can monitor and control file and directory access, which provides an additional security layer.
+
This feature helps prevent unauthorized code execution and enhances system integrity.
You can enable or disable `fapolicyd` on both {ProjectServer} and {SmartProxyServer} at any time, depending on your security requirements.
+
For additional information about `fapolicyd`, including instructions on enabling it, see {RHELDocsBaseURL}9/html/security_hardening/assembly_blocking-and-allowing-applications-using-fapolicyd_security-hardening[Blocking and allowing applications by using fapolicyd] in _{RHEL}{nbsp}9 Security hardening_.

Security-Enhanced Linux (SELinux)::
{Project} supports SELinux, a mandatory access control system that restricts system access and reduces the risk of security vulnerabilities.
SELinux enforces policies that define which processes can access specific system resources, providing protection against unauthorized access and exploitation.
+

> Running {Project} with SELinux in enforcing mode is recommended to maintain a high-security environment.

Reviewer: suggestion on the wording: "is recommended" -> "{Team} recommends running {Project} ..."

It ensures that access control policies are strictly enforced, minimizing the risk of privilege escalation or unauthorized actions.
+

> Using SELinux is required for {Project} Server to be installed on your environment.

Reviewer: ```suggestion
Using SELinux is required for {ProjectServer} to be installed on your environment.
```

This is confusing to me, esp. with line 39 "{Project} support SELinux". Should this say "{Project} requires SELinux"? Or maybe just drop this sentence?

In general, I would be OK with **not** documenting the less secure option aka. no mandatory SELinux.
